src/hash.rs

```diff
@@ -1,123 +1,299 @@
+        assert_eq!("sha512".parse(), Ok(HashAlgorithm::Sha512));
+        assert_eq!("sha3-256".parse::<HashAlgorithm>(),
+                   Err(HashError::new(HashErrorKind::InvalidHashAlgorithm,
+                                      String::from("Invalid hash algorithm 'sha3-256'"))));
     }
 
     #[test]
-    fn error_on_invalid_hash() {
-        let hash = "123456";
-        let res = hash_parse(&hash);
-        assert!(res.is_err());
-        assert_eq!(res.unwrap_err().to_string(), "invalid hash '123456'");
+    fn can_deparse_hash() {
+        let h = Hash {
+            algorithm: HashAlgorithm::Md5,
+            value: String::from("123"),
+        };
+        assert_eq!(h.to_string(), "md5:123");
     }
 
     #[test]
-    fn validates_hash() {
-        let hash = "badhash";
-        let res = validate_hash("tests/example", hash, "1234".as_bytes());
-        assert!(res.is_err());
-        assert_eq!(res.unwrap_err().to_string(), "invalid hash 'badhash'");
+    fn can_parse_hash() {
+        assert_eq!(
+            "md5:1234".parse(),
+            Ok(Hash {
+                algorithm: HashAlgorithm::Md5,
+                value: String::from("1234")
+            })
+        );
+        assert_eq!(
+            " sha256:abcde".parse(),
+            Ok(Hash {
+                algorithm: HashAlgorithm::Sha256,
+                value: String::from("abcde")
+            })
+        );
+        assert_eq!("md51234".parse::<Hash>(),
+                   Err(HashError::new(HashErrorKind::InvalidHashFormat,
+                                     String::from("Invalid hash format 'md51234'"))));
+        assert_eq!(
+            "sha666:1234".parse::<Hash>(),
+            Err(HashError::new(HashErrorKind::InvalidHashAlgorithm,
+                              String::from("Invalid hash algorithm 'sha666'"))));
     }
 
     #[test]
-    fn validates_hash_algorithm() {
-        let hash = "md5:e9aa9f2212aba6fba4464212800a2927afa02eda688cf13131652da307e3d7c1";
-        let res = validate_hash("tests/example", hash, "1234".as_bytes());
-        assert!(res.is_err());
-        assert_eq!(res.unwrap_err().to_string(), "Hash algorithm md5 does not match config. Expected sha256");
+    fn can_hash_simple_data() {
+        /*
+        > orderly2:::hash_data("1234", "md5")
+        [1] "md5:81dc9bdb52d04dc20036dbd8313ed055"
+        > orderly2:::hash_data("1234", "sha1")
+        [1] "sha1:7110eda4d09e062aa5e4a390b0a572ac0d2c0220"
+        > orderly2:::hash_data("1234", "sha256")
+        [1] "sha256:03ac674216f3e15c761ee1a5e255f067953623c8b388b4459e13f978d7c846f4"
+        > orderly2:::hash_data("1234", "sha384")
+        [1] "sha384:504f008c8fcf8b2ed5dfcde752fc5464ab8ba064215d9c5b5fc486af3d9ab8c81b14785180d2ad7cee1ab792ad44798c"
+        > orderly2:::hash_data("1234", "sha512")
+        [1] "sha512:d404559f602eab6fd602ac7680dacbfaadd13630335e951f097af3900e9de176b6db28512f2e000b9d04fba5133e8b1c6e8df59db3a8ab9d60be4b97cc9e81db"
+         */
```

Comment:
Bit of old comment?

Reply:
this was showing where the values come from, just to explain the random string literals